OpenDreamShared/Resources/DMIParser.cs

```diff
@@ -44,6 +45,80 @@ public sealed class ParsedDMIDescription {
                 };
             }
 
+            public void InsertIcon(ParsedDMIDescription icon, string icon_state, AtomDirection? dir, int? frame, float? delay)
+            {
+                List<ParsedDMIState> states;
+                if (icon_state is null)
+                {
+                    states = icon.States.Values.ToList();
+                }
+                else
+                {
+                    states = new List<ParsedDMIState>(1) { icon.States[icon_state] };
+                }
```

Comment:
Avoid an unnecessary `ToList()`
```suggestion
                IEnumerable<ParsedDMIState> states;
                if (icon_state is null) {
                    states = icon.States.Values;
                } else {
                    states = new ParsedDMIState[] { icon.States[icon_state] };
                }
```